src/anndata/_core/anndata.py

```diff
@@ -1014,28 +1039,12 @@ def __getitem__(self, index: Index) -> AnnData:
         return AnnData(self, oidx=oidx, vidx=vidx, asview=True)
 
     def _remove_unused_categories(
-        self, df_full: pd.DataFrame, df_sub: pd.DataFrame, uns: dict[str, Any]
+        self,
+        df_full: pd.DataFrame,
+        df_sub: pd.DataFrame,
+        uns: dict[str, Any],  # types are wrong now...
     ):
-        for k in df_full:
-            if not isinstance(df_full[k].dtype, pd.CategoricalDtype):
-                continue
-            all_categories = df_full[k].cat.categories
-            with pd.option_context("mode.chained_assignment", None):
-                df_sub[k] = df_sub[k].cat.remove_unused_categories()
-            # also correct the colors...
-            color_key = f"{k}_colors"
-            if color_key not in uns:
-                continue
-            color_vec = uns[color_key]
-            if np.array(color_vec).ndim == 0:
-                # Make 0D arrays into 1D ones
-                uns[color_key] = np.array(color_vec)[(None,)]
-            elif len(color_vec) != len(all_categories):
-                # Reset colors
-                del uns[color_key]
-            else:
-                idx = np.where(np.isin(all_categories, df_sub[k].cat.categories))[0]
-                uns[color_key] = np.array(color_vec)[(idx,)]
+        _remove_unused_categories(df_full, df_sub, uns)
```

Comment:
is it necessary to keept the method if you make it into a function anyway? why not just change all call sites to the function?